lib/Conversion/TorchToStd/TorchToStd.cpp

```diff
@@ -216,8 +218,17 @@ class ConvertAtenAnyBoolOp : public OpConversionPattern<AtenAnyBoolOp> {
             op, "only support constant bool input list elements");
       inputListBool.push_back(cst);
     }
-    bool result = llvm::any_of(
+    
+    bool result = true;
+    if (auto getBoolOp = llvm::dyn_cast<AtenAnyBoolOp>(&op)) {
+       result = llvm::any_of(
         inputListBool, [](bool inputListElem) { return inputListElem; });
+    }
+    else if (auto getBoolOp = llvm::dyn_cast<AtenAllBoolOp>(&op)){
+       result = llvm::all_of(
+        inputListBool, [](bool inputListElem) { return inputListElem; });
+    }
```

Comment:
I'm usually a bit reluctant to have `dyn_cast` inside of a template because they're design choices that conflict on their goals. One of the main benefits of templates is to allow polymorphism without any runtime overhead. However, a `dyn_cast` does require runtime overhead, so it counteracts the benefits of the template.

For what you're trying to do here there are a couple of common design patterns. The first is to do something like what is used in the https://github.com/llvm/torch-mlir/blob/main/lib/Conversion/TorchToTosa/TorchToTosa.cpp file. Namely, you first create a base template class

```cpp
template <typename AtenOpT>
class ConvertAtenAnyOrAllBoolOp : public OpConversionPattern<AtenOpT> {
  // Method to be implemented by classes that inherit from this class
  virtual bool reduceBoolArray(ArrayRef<bool> boolArray) const;
  // ...
  LogicalResult matchAndRewrite(AtenOpT op, ...) const override {
    // ...
    bool result = reduceBoolArray(inputListBool);
    // ...
  }
}

class ConvertAtenAnyBoolOp : public ConvertAtenAnyOrAllOp<AtenAnyBoolOp> {
  bool reduceBoolArray(ArrayRef<bool> boolArray) const override {
    return llvm::any_of(boolArray, [](bool inputListElem) { return inputListElem; });
  }
}

// same for `AtenAllBoolOp`
```

The other pattern is to create a static helper function instead of a base class. You would do something like 

```cpp
static LogicalResult computeAnyOrAllBoolOp(// whatever argument are needed,
                 function_ref<bool(ArrayRef<bool>, function_ref<bool(bool)>)> reduceBoolArray) {
  return reduceBoolArray(inputListBool, [](bool inputListElem) { return inputListElem; });
}
```

Then define one class for each op and the `matchAndRewrite` body will just be the helper function.

Happy to help with any questions